Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -273,31 +284,60 @@ namespace AzToolsFramework::Prefab
+                previousFocusedInstance->get().GetEntities(
+                    [&](AZStd::unique_ptr<AZ::Entity>& entity) -> bool
+                {
+                    entities.push_back(entity->GetId());
+                    return true;
+                }
+                );
+                entities.push_back(previousFocusedInstance->get().GetContainerEntityId());
             }
-            containerEntities.push_back(focusedInstance->get().GetContainerEntityId());
+            entities.push_back(focusedInstance->get().GetContainerEntityId());
+            focusedInstance->get().GetEntities(
+                [&](AZStd::unique_ptr<AZ::Entity>& entity) -> bool
+            {
+                entities.push_back(entity->GetId());
+                return true;
+            });
 
-            m_readOnlyEntityQueryInterface->RefreshReadOnlyState(containerEntities);
+            m_readOnlyEntityQueryInterface->RefreshReadOnlyState(entities);
         }
 
         // Refresh path variables.
         RefreshInstanceFocusPath();
 
         // Open all container entities in the new path.
         SetInstanceContainersOpenState(m_rootAliasFocusPath, true);
+        // Set open state on all nested instances in the new focus subtree based on edit scope.
+        SetInstanceContainersOpenStateOfAllDescendantContainers(
+            GetInstanceReference(m_rootAliasFocusPath), m_prefabEditScope == PrefabEditScope::SHOW_NESTED_INSTANCES_CONTENT);
 
-        PrefabFocusNotificationBus::Broadcast(&PrefabFocusNotifications::OnPrefabFocusChanged);
+        AZ::EntityId previousFocusedInstanceContainerEntityId = previousFocusedInstance.has_value() ?
+            previousFocusedInstance->get().GetContainerEntityId() : AZ::EntityId();
+        AZ::EntityId currentFocusedInstanceContainerEntityId = focusedInstance.has_value() ?
+            focusedInstance->get().GetContainerEntityId() : AZ::EntityId();
+        if (previousFocusedInstanceContainerEntityId != currentFocusedInstanceContainerEntityId)
+        {
+            PrefabFocusNotificationBus::Broadcast(&PrefabFocusNotifications::OnPrefabFocusChanged,
+                previousFocusedInstanceContainerEntityId, currentFocusedInstanceContainerEntityId);
+        }
 
         // Force propagation on both the previous and the new focused instances to ensure they are represented correctly.
+        // The most common operation is focusing a prefab instance nested in the currently focused instance.
+        // Queuing the previous focus before the new one saves some time in the propagation loop on average.
         if (previousFocusedInstance.has_value())
         {
             m_instanceUpdateExecutorInterface->AddInstanceToQueue(previousFocusedInstance);
         }
-        m_instanceUpdateExecutorInterface->AddInstanceToQueue(focusedInstance);
+        if (focusedInstance.has_value())
+        {
+            m_instanceUpdateExecutorInterface->AddInstanceToQueue(focusedInstance);
+        }
```

Comment:
nit: `focusedInstance` is guaranteed to have a value at this point, as we have an early out on line 258. So this check is redundant.